src/kinship/hierarchy-creator.ts

```diff
@@ -0,0 +1,321 @@
+import * as d3 from 'd3';
+import { DataProvider, Indi, Fam } from '../api';
+import { ChildNodes, TreeNode, LinkType, otherSideLinkType } from './api';
+import { IdGenerator } from '../id-generator';
+import { nonEmpty, flatten } from '../utils';
+
+
+const FAM_ID_CHAR = "F";
```

Comment:
Checking the first character of the identifier is not very reliable. I'd prefer to explicitly pass something like:
interface FamOrIndiId {
  indiId?: string;
  famId?: string;
}

Reply:
Done